packages/devp2p/src/eth/index.ts

```diff
@@ -263,6 +263,17 @@ export class ETH extends EventEmitter {
         this._latestBlock = latestBlock
       }
       const forkHashB = Buffer.from(this._forkHash.substr(2), 'hex')
+
+      // EIP-3675: Set merge FORK_NEXT to mergeForkBlock
+      if (this._peer._common.hardfork() === 'london' && this._nextForkBlock.eqn(0)) {
+        const mergeForkBlock = (
+          this._peer._common.hardforks().filter((h) => h.name === 'merge') as any
+        )[0]?.mergeForkBlock
+        if (mergeForkBlock) {
+          this._nextForkBlock = new BN(mergeForkBlock)
```

Comment:
I am not sure if we might be able to circumvent all this extra code by defining an extra hardfork `preMerge` or similar in `Common` and then this `preMerge` HF gets the block number and the `Merge` HF itself gets the difficulty switch.

Then all the devp2p switches should work automatically I guess and on first thought this shouldn't have any side effects (checks like `if HF >= london do x`) should still work, also all Merge-related stuff should still be on `if HF >= Merge`.

Reply:
that's a great idea, I was struggling on how to conceptualize this, defining a "preMerge" hardfork for this purpose sounds like it would work well, I will give it a try.

Reply:
done in https://github.com/ethereumjs/ethereumjs-monorepo/pull/1565/commits/e0d436eddeea47c7ac8aeee29ff83d11f10582a7, synced to terminal block within a few mins! with very quick block execution

<img width="1513" alt="Screen Shot 2022-02-01 at 10 22 32 PM" src="https://user-images.githubusercontent.com/22116/152104945-fc2f3d31-7613-4964-b49f-f9f4581246a2.png">